mobile_config/src/gateway_info.rs

```diff
@@ -33,11 +33,15 @@ impl From<GatewayInfoProto> for GatewayInfo {
     fn from(info: GatewayInfoProto) -> Self {
         let metadata = if let Some(metadata) = info.metadata {
             u64::from_str_radix(&metadata.location, 16)
-                .map(|location| GatewayMetadata { location })
+                .map(|location| GatewayMetadata {
```

Comment:
not sure this implementation makes sense any longer since a mobile hotspot should _always_ have a device type the struct going forward should probably be
```
GatewayInfo {
    address: PublicKeyBinary,
    metadata: GatewayMetadata {
        location: Option<u64>,
        device_type: DeviceTypeEnum
    }
}
```
if we wanna minimize definitions we could probably pull the device type enum in from the HPL since it's already a dependency of Oracles but at the very least I'd make this an enum rather than a string